prov/efa/src/rdm/efa_rdm_pke_utils.c

```diff
@@ -229,23 +228,62 @@ int efa_rdm_pke_queued_copy_payload_to_hmem(struct efa_rdm_pke *pke,
 	return efa_rdm_ep_flush_queued_blocking_copy_to_hmem(ep);
 }
 
-/* @brief copy data in pkt_entry to CUDA memory
+/*
+ * @brief check which copy methods are available
  *
  * There are 3 ways to copy data to CUDA memory. None of them is guaranteed to
  * be available:
  *
- * gdrcopy, which is avaibale only when cuda_is_gdrcopy_enabled() is true
+ * localread copy, which is available only when p2p is supported by device, and
+ * device support read.
+ *
+ * gdrcopy, which is available only when cuda_is_gdrcopy_enabled() is true.
  *
- * cudaMemcpy, which is available only when endpoint is permitted to call CUDA api
+ * cudaMemcpy, which is available only when endpoint is permitted to call CUDA
+ * api.
  *
- * localread copy, which is available only when p2p is supported by device, and device support read.
+ * @param[in]			ep, desc
+ * @param[in,out]		local_read_available, cuda_memcpy_available,
+ * gdrcopy_available
+ * @return		On success, return 0
+ * 				On failure, return libfabric error code
+ */
+int efa_rdm_pke_get_available_copy_methods(struct efa_rdm_ep *ep,
+					   struct efa_mr *desc,
+					   bool *local_read_available,
+					   bool *cuda_memcpy_available,
+					   bool *gdrcopy_available)
```

Comment:
We can `restrict`-qualify the parameters here to avoid any aliasing quirks

```suggestion
					   bool *restrict local_read_available,
					   bool *restrict cuda_memcpy_available,
					   bool *restrict gdrcopy_available)
```